### PR TITLE
Fixed #24291 - Fixed migrations with unused swappable models

### PR DESCRIPTION
**Pull request by knbk**
Changed ModelState from_model to use an empty list if the model was an unused
swappable model.

### CODE REVIEW

**Comment by knbk** [comment]
Flake8 doesn't fail on my local system. Could someone give me any pointers on how to fix it?

**Comment by MarkusH** [comment]
Here you go:
```
./tests/migrations/test_state.py:568:9: E303 too many blank lines (2)
```

**Comment by knbk** [comment]
I just double-checked and PEP8 allows single lines to separate blocks. Why did flake8 mark that single blank line as an error? Anyway, it works now.

**Comment by MarkusH** [comment]
buildbot, test this please.

**Comment by MarkusH** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
Could you also make the use of quotes consistent (single quote everywhere should be fine, I think)

**Comment by knbk** [comment]
db/migrations/state.py seems to use double quotes almost everywhere, so I used them there as well. tests/migrations/test_state.py was more 50/50 so I switched to consistently using single quotes in the new test.

**Comment by MarkusH** [comment]
buildbot, test this please.

**Comment by MarkusH** [comment]
With this patch Django creates a migrations for `auth` that drops the managers from `User` and `Permission`.

**Comment by collinanderson** [comment]
buildbot, test this please

**Comment by collinanderson** [comment]
buildbot, test this please

File: tests/migrations/test_state.py [hunk 1]
```diff
@@ -595,6 +595,62 @@ def test_manager_refer_correct_model_version(self):
         self.assertIsNot(old_model.food_mgr.model, new_model.food_mgr.model)
         self.assertIsNot(old_model.food_qs.model, new_model.food_qs.model)
 
+    @override_settings(TEST_SWAPPABLE_MODEL='migrations.SomeFakeModel')
+    def test_create_swappable(self):
+        """
+        Tests making a ProjectState from an Apps with a swappable model
+        """
+
```
Comment (timgraham) [inline tests/migrations/test_state.py]: no blank line needed

File: tests/migrations/test_state.py [hunk 2]
```diff
@@ -595,6 +595,62 @@ def test_manager_refer_correct_model_version(self):
...
+    @override_settings(TEST_SWAPPABLE_MODEL='migrations.SomeFakeModel')
+    def test_custom_manager_swappable(self):
+        """
+        Tests making a ProjectState from unused models with custom managers
+        """
+
+        new_apps = Apps(['migrations'])
+
+        class Food(models.Model):
+
```
Comment (timgraham) [inline tests/migrations/test_state.py]: no blank line needed

**Comment by timgraham** [comment]
You can squash commits, thanks.

**Comment by MarkusH** [comment]
buildbot, test this please.

**Comment by MarkusH** [comment]
Test failure due to failed checkout

**Comment by MarkusH** [comment]
Merged in 15dc8d1c9d3697170a2c59ecaa7a2b4ba58f5990. Thanks.

**Comment by MarkusH** [comment]
I slightly changed the test cases to not use an entire `ProjectState` but just `ModelState.from_model()` and thus moved them into `ModelStateTests`.

**Pull request closed without merging**